contribute/architecture/state.md

```diff
@@ -0,0 +1,142 @@
+# State
+
+Grafana uses a database to persist settings between restarts. In fact, if you don't specify one, Grafana creates a [SQLite3](https://www.sqlite.org/) database file on your local disk. This guide explains how to store and retrieve data from the database.
+
+Grafana supports the following databases:
```

Comment:
Maybe link to https://grafana.com/docs/installation/requirements/#database and/or add some additional information to the requirements documentation that can be linked to?